.buildkite/steps/tests.sh

```diff
@@ -8,14 +8,18 @@ go install gotest.tools/gotestsum@v1.8.0
 
 go install github.com/buildkite/test-engine-client@v1.5.0-rc.3
 
+COVERDIR="/tmp/coverdir"
+mkdir "$COVERDIR"
+
 echo '+++ Running tests'
 export BUILDKITE_TEST_ENGINE_SUITE_SLUG=buildkite-agent
 export BUILDKITE_TEST_ENGINE_TEST_RUNNER=gotest
 export BUILDKITE_TEST_ENGINE_RESULT_PATH="junit-${BUILDKITE_JOB_ID}.xml"
 export BUILDKITE_TEST_ENGINE_RETRY_COUNT=1
-export BUILDKITE_TEST_ENGINE_TEST_CMD="gotestsum --junitfile={{resultPath}} -- -count=1 -coverprofile=cover.out $@ {{packages}}"
+export BUILDKITE_TEST_ENGINE_TEST_CMD="gotestsum --junitfile={{resultPath}} -- -count=1 -cover -test.gocoverdir=$COVERDIR $@ {{packages}}"
 
 test-engine-client
 
 echo 'Producing coverage report'
+go tool covdata textfmt -i "$COVERDIR" -o cover.out
```

Comment:
so as we're parallising we'll need to merge it for the parallel runs too using artifacts and a dependant step I think?

Reply:
ahhh true! let me pull this back to draft.